### PR TITLE
updated README.md

### PR DESCRIPTION
**Pull request by kdarkhan**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,52 @@
...
+### ConfigurationBuilder
+create a new instance of ConfigurationBuilder
+```
+var ConfBuilder = require('context-config').ConfigurationBuilder;
+var builder = new ConfBuilder(emitter);
+```
+call build method to load the config.
+```
+builder.build(context, ref)
+.then(function(config){
```
Comment (inexplicable) [inline README.md:30]: please mention it's returning a promise, therefore, then, fail, fin could all be chained, "then" handles the success case

File: README.md [hunk 2]
```diff
@@ -1,4 +1,52 @@
...
+|`build(context, ref)`| `ref` object lists properties of the app whose configs we want to load (can contain properties domain, target, project, config, and version). Returns a promise.
+
+
+
+## Usage
+### ConfigurationBuilder
+create a new instance of ConfigurationBuilder
+```
+var ConfBuilder = require('context-config').ConfigurationBuilder;
+var builder = new ConfBuilder(emitter);
```
Comment (inexplicable) [inline README.md]: ConfBuilder is a bit confusing, just use ConfigurationBuilder, for the class name, that is more conventional

File: README.md [hunk 3]
```diff
@@ -1,4 +1,52 @@
...
+create a new instance of ContextConfiguration
+```
+var ContextConfiguration = require('context-config').ContextConfiguration;
+var config = new ContextConfiguration([],[]);
+```
+append new value to the config
+```
+config.append({
+    "key":"k1",
+    "context": {"site":"en-US"},
```
Comment (inexplicable) [inline README.md:56]: please separate with/without context cases, without context being the 1st one to explain as it's simpler

**Comment by kdarkhan** [comment]
i've made the changes, let me know if anything else needs to be done.

**Pull request merged**